ISSUE-LABELLING.md

```diff
@@ -0,0 +1,234 @@
+
+* comp/floodsub
+* comp/gossipsub
+* comp/kad-dht
+* comp/noise
+* comp/quic
+* comp/secio
+* comp/tcp
+* comp/tls
+* comp/webrtc
+* comp/wss
+* comp/yamux
+
+### Difficulty (`dif/`)
+
+> Subjectively estimates the difficulty of resolution of an issue, based on
+> prior expertise about libp2p and its constructs, p2p networking, systems
+> programming, etc.
+
+Values and calibration guidance:
+
+* dif/trivial: Can be confidently tackled by newcomers, who are widely
+  unfamiliar with libp2p.
+* dif/easy: A prexisting libp2p user should be able to pick this up.
+* dif/medium: Prior experience in having developed libp2p modules is likely
+  helpful.
+* dif/hard: Suggests that having worked on the specific component affected
+  by this issue is important.
+* dif/expert: Requires extensive knowledge of the history, implications,
+  ramifications of the issue, as well as deep understanding of the libp2p stack
+  and experience with systems and network programming.
+
+### Size (`size/`)
+
+> Estimates the amount of work. This axis is distinct to the complexity, e.g.
+> something can be easy but require a lot of work, or vice versa. 
+
+Values and calibration guidance:
+
+* size/XS: minutes.
+* size/S: several hours.
+* size/M: one day.
+* size/L: 2-3 days.
+* size/XL: 1 week.
+* size/XXL: > 1 week.
+
+### Kind (`kind/`)
+
+> Describes the nature of the issue.
+
+* kind/bug
+* kind/improvement
+* kind/tracking-issue
+* kind/question
+* kind/enhancement
+* kind/architecture
+* kind/discussion
+
+### Impact (`impact/`)
+
+> Contextualises the impact, and therefore the perceived/derived severity, of
+> the issue.
+
+* impact/regression
+* impact/api-breakage
+* impact/quality
+* impact/dx
+* impact/test-flakiness
+
+### Topic (`topic/`)
+
+> Classifies issues pertaining to particular themes. Topics can be created to
+> refer to epics, larger bodies of work, or ad-hoc endeavours.
+
+* topic/interoperability
+* topic/docs
+* topic/infra
+
+## Execution labels
+
+These labels describe some aspect of the execution on the issue. These labels
+are dynamic; they change as a result of planning, analysis, scoping, and
+generally applying the development workflow. They are usually set by
+maintainers.
+
+### Priority (`Pn`)
+
+> Indicates the current implementation priority; used for planning and
+> roadmapping purposes.
+
+* P0: Critical.
+* P1: High.
+* P2: Medium.
+* P3: Low.
+
+### Hints/advisories (`hint/`)
+
+> Surfaces observations and facts about issues, facilitating progress to be
+> made.
+
+* hint/good-first-issue
+* hint/needs-contributor
+* hint/needs-participation
+* hint/needs-decision
+* hint/needs-triage
+* hint/needs-analysis
+* hint/needs-author-input
+* hint/needs-team-input
+* hint/needs-community-input
+* hint/needs-review
+* hint/needs-help
+
+### Status (`status/`)
+
+> Marks the current status of the issue. 
+
+* status/done
+* status/deferred
+* status/in-progress
+* status/blocked
+* status/inactive
+* status/waiting
+* status/rotten
```

Comment:
`status/rotten` isn't quite clear to me. Are these stale issues? Won't fix issues? Or something else? It sounds like this applies to stale issues, and if so I'd recommend changing to that verbiage, `status/stale`.

Reply:
I suspect this category may be hiding two finer categories within it:

* status: which stage in the workflow is this issue in? If we ever create a flowchart of the process by which issues progress, each status label would map to a box.
* outcome or resolution: when the status is a terminal one (e.g. `done`), what is the result? `outcome/wont-fix`, `outcome/implemented`.

But for the sake of starting simple, I advise not to break this up until necessary.

To me `status/rotten` denotes a terminal state (user should expect no action on it). `status/stale` is lighter. You would throw away rotten bread, but with stale bread you could make [gazpacho](https://en.wikipedia.org/wiki/Gazpacho).